pkg/ingester/ingester_test.go

```diff
@@ -80,6 +80,150 @@ func mustNewActiveSeriesCustomTrackersConfigFromMap(t *testing.T, source map[str
 	return m
 }
 
+func TestIngester_StartPushRequest(t *testing.T) {
+	instanceLimits := &InstanceLimits{}
+	pushReqState := &pushRequestState{
+		requestSize:                  10,
+		acquiredCircuitBreakerPermit: true,
+	}
+	ctx := context.Background()
+	ctxWithPushReqState := context.WithValue(ctx, pushReqCtxKey, pushReqState)
+
+	type testCase struct {
+		ctx                            context.Context
+		failingIngester                bool
+		cbOpen                         bool
+		cbInitialDelay                 time.Duration
+		instanceLimitReached           bool
+		expectedStatus                 bool
+		expectedFinishPushRequestCalls int
```

Comment:
Instead of checking `expectedFinishPushRequestCalls` (which isn't even checked), tests could verify number of inflight requests (0 on failure, 1 on success).

Reply:
I have modified the test, but it is not true that the number of inflight requests will be 0 on failure and 1 on success: if we call `startPushRequest` for the second time on the same request, it will not cause a failure, but it will also not increase the number of inflight push requests.